src/HeapAllocator.jl

```diff
@@ -0,0 +1,308 @@
+module HeapAllocator
+
+@eval macro $(Symbol("const"))(field)
+    if VERSION >= v"1.8.0-DEV.1148"
+        Expr(:const, esc(field))
+    else
+        return esc(field)
+    end
+end
```

Comment:
Worth putting in Compat.jl? 
Ref: https://github.com/JuliaLang/Compat.jl/issues/800